schema-registry/core/src/main/java/com/hortonworks/registries/schemaregistry/DefaultSchemaRegistry.java

```diff
@@ -533,7 +528,29 @@ public SchemaIdVersion mergeSchemaVersion(Long schemaVersionId, SchemaVersionMer
         Collection<AggregatedSchemaBranch> aggregatedSchemaBranches = new ArrayList<>();
         for (SchemaBranch schemaBranch : getSchemaBranches(schemaName)) {
             Long rootVersion = schemaBranch.getName().equals(SchemaBranch.MASTER_BRANCH) ? null: schemaVersionLifecycleManager.getRootVersion(schemaBranch).getId();
-            aggregatedSchemaBranches.add(new AggregatedSchemaBranch(schemaBranch, rootVersion, getAllVersions(schemaBranch.getName(), schemaName)));
+            Collection<SchemaVersionInfo> schemaVersionInfos = getAllVersions(schemaBranch.getName(), schemaName);
+            schemaVersionInfos.stream().forEach(schemaVersionInfo -> {
+                SchemaVersionLifecycleContext context = null;
+                try {
+                    context = schemaVersionLifecycleManager.createSchemaVersionLifeCycleContext(schemaVersionInfo.getId(), SchemaVersionLifecycleStates.INITIATED);
```

Comment:
Effectively, there will be a DB call for each schema version as `schemaVersionLifecycleManager.createSchemaVersionLifeCycleContext` internally needs to make that DB call.